product_extra_image_zipped_download/models/product_image.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import mimetypes
+
+from odoo import api, fields, models
+from odoo.tools.mimetypes import get_extension, guess_mimetype
+
+
+class ProductImage(models.Model):
+    _inherit = "product.image"
+
+    img_name = fields.Char("Image Name")
+    update_imge_name = fields.Boolean()
+
+    @api.model
+    def create(self, vals):
+        if not vals.get("img_name"):
+            name = vals.get("name")
+            image_data = base64.b64decode(vals["image_1920"])
+            mimetype = guess_mimetype(image_data)
+            extension = get_extension(name)
+            if not extension:
+                extension = mimetypes.guess_extension(mimetype)
+            vals["img_name"] = f"{name}{extension}" if extension else name
```

Comment:
```suggestion
            vals["img_name"] = f"{id}{name}{extension}" if extension else f"{id}{name}"
            vals["update_imge_name"] = True
```
I want to make the img_name unique by adding the ID at the beginning.